docs/design/StratisSoftwareDesign.lyx

```diff
@@ -762,905 +744,1284 @@ Handle million+ filesystems in a pool and updates without writing to each
-All methods return at least a pair, consisting of a non-negative integer
- return code and a message string.
- A method may return more values; these precede the return code.
-\begin_inset Note Comment
-status collapsed
+\begin_layout Enumerate
+Cache redundancy level: raid1
+\end_layout
 
-\begin_layout Plain Layout
-houldn't all the objects have some type of UUID?
+\begin_layout Enumerate
+Write-through caching enabled
 \end_layout
 
-\end_inset
+\begin_layout Enumerate
+thin/cache metadata validation/check (i.e.
+ call thin_check & cache_check)
+\end_layout
 
+\begin_layout Enumerate
+Quotas
+\end_layout
 
-\begin_inset Note Comment
-status collapsed
+\begin_layout Enumerate
+Blockdev resize (larger)
+\end_layout
 
-\begin_layout Plain Layout
-If you utilize the standard dbus interfaces you can leverage ObjectManager
- and it's signals for objects being created and removed.
- You can also leverage the properties interface to get signals when properties
- of the objects change as well.
+\begin_layout Enumerate
+Spares
 \end_layout
 
-\end_inset
+\begin_layout Subsection
+Stratis version 3.0
+\end_layout
 
+\begin_layout Standard
+Rough ZFS feature parity.
+ New DM features needed.
+\end_layout
 
-\begin_inset Note Comment
-status collapsed
+\begin_layout Enumerate
+Send/Receive Integrity checking (w/ self-healing only if on raid)
+\end_layout
 
-\begin_layout Plain Layout
-Have you thought about monitoring issues? Beyond just logging, it would
- be good to have a mechanism to query cache hit rates, I/O amounts/speeds,
- identity of all constituent devices, whatever data might be available from
- the lower level md/kernel stats.
+\begin_layout Enumerate
+Raid scrub
 \end_layout
 
-\end_inset
+\begin_layout Enumerate
+Compression
+\end_layout
 
+\begin_layout Enumerate
+Encryption
+\end_layout
 
+\begin_layout Enumerate
+Dedupe[aq]
 \end_layout
 
-\begin_layout Subsection
-Manager
+\begin_layout Enumerate
+Raid write log (on ssd? To eliminate raid write hole)
 \end_layout
 
-\begin_layout Subsubsection
-Overview
+\begin_layout Subsection
+Stratis version 4.0
 \end_layout
 
 \begin_layout Standard
-Manager is the top-level interface.
- It makes available meta-information about stratisd, high-level methods,
- and general-purpose accessor methods.
+Future features and evolution
 \end_layout
 
-\begin_layout Subsubsection
-Introspection Data
-\end_layout
-
-\begin_layout Standard
-\begin_inset Box Frameless
-position "t"
-hor_pos "c"
-has_inner_box 1
-inner_pos "t"
-use_parbox 0
-use_makebox 0
-width "100col%"
-special "none"
-height "1in"
-height_special "totalheight"
-thickness "0.4pt"
-separation "3pt"
-shadowsize "4pt"
-framecolor "black"
-backgroundcolor "none"
-status collapsed
+\begin_layout Enumerate
+Change a pool’s redundancy level
+\end_layout
 
-\begin_layout Plain Layout
-\begin_inset CommandInset include
-LatexCommand verbatiminput
-filename "manager_introspection.txt"
+\begin_layout Enumerate
+Boot from a filesystem
+\end_layout
 
```

Comment:
Since you're removing the reference to the file, you should remove the file as well.